src/components/TodoModal/TodoModal.tsx

```diff
@@ -15,31 +28,41 @@ export const TodoModal: React.FC = () => {
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              {`Todo #${selectedTodo?.id}`}
             </div>
 
             {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
             <button
               type="button"
               className="delete"
               data-cy="modal-close"
+              onClick={() => {
+                setSelectedTodo(null);
+              }}
             />
           </header>
 
           <div className="modal-card-body">
             <p className="block" data-cy="modal-title">
-              quis ut nam facilis et officia qui
+              {selectedTodo && selectedTodo.title}
             </p>
 
             <p className="block" data-cy="modal-user">
-              {/* <strong className="has-text-success">Done</strong> */}
-              <strong className="has-text-danger">Planned</strong>
-
+              {selectedTodo ? (
+                <strong
+                  className={
+                    selectedTodo.completed
+                      ? 'has-text-success'
+                      : 'has-text-danger'
+                  }
+                >
+                  {selectedTodo.completed ? 'Done' : 'Planned'}
+                </strong>
+              ) : null}
               {' by '}
-
-              <a href="mailto:Sincere@april.biz">
-                Leanne Graham
-              </a>
+              {selectedUser ? (
+                <a href={`mailto:${selectedUser.email}`}>{selectedUser.name}</a>
+              ) : null}
```

Comment:
```suggestion
              {selectedUser && (
                <a href={`mailto:${selectedUser.email}`}>{selectedUser.name}</a>
              )}
```

can be more simplify